### ISSUE TITLE
Playback not working when saving frames

### ISSUE THREAD

**Issue by Arohdin** [OpenSpace/OpenSpace#2593]
There seem to be some issue with saving frames during playback of recordings.
Sometimes it works as intended while other times the playback does not work properly and nothing happens.
More specifically the playback does not start and it just stands still (UI switches to playback mode).

Issue has been observed by both me and Alex on different computers.
I've also had the same issue on multiple different versions of OpenSpace (17, 18.2 and current master branch).

Playback without saving frames works as intended and the playback plays fine.

https://user-images.githubusercontent.com/9076357/231154386-69a5a370-a272-4dee-9723-d567808d3123.mp4

**Comment by GPayne** [comment]
This happens when rendering with frames and when a globebrowsing planet is in the view. The rendering will wait indefinitely because the playback time will only be incremented when RenderableGlobe::renderedWithDesiredData() returns true when all tiles are loaded.
https://github.com/OpenSpace/OpenSpace/blob/9be88fdea8af74f5568ca2c5f393aadd465bcc4d/src/interaction/sessionrecording.cpp#L1866

**Comment by alexanderbock** [comment]
Fixing the issue why it no longer reports that is ready is definitely necessary.  It feels like an error we have introduced recently with some changes in the tileproviders.

How about also adding an option to the UI to turn that feature on or off? Thoughts?

**Comment by WeirdRubberDuck** [comment]
Makes sense to me to have a setting in the UI. I don't see why we would always want to wait for tiles to be loaded.  Also, having a setting for it allows us to communicate to the user that this is something that is possible and that might affect the playback behaviors used through the UI

**Comment by GPayne** [comment]
This is happening at certain dates with the Earth VIIRS globebrowsing. I reproduced the problem at the original date/time as first posted (both windows and linux).
The following video shows a recording being made while the patch of imagery isn't loaded, and then attempting to playback (0:12) with "output frames" selected just hangs. Next, the camera is moved around until the imagery loads for whatever reason, a new recording is made (0:39), and then playing back that recording works until it gets to the point where the patch goes blank again and then it hangs until manually stopped (0:59).
I've observed this same issue when setting the date to April 12th/13th of 2023, where there is again a similar blank patch of imagery. Coincidentally, this was when the issue was first posted and when others were trying to replicate it.

[2593_3.webm](https://user-images.githubusercontent.com/3966290/234432159-e9df3b45-d3f0-4e21-962b-65327ef2b91d.webm)
